src/main/java/io/confluent/connect/elasticsearch/ElasticsearchSinkTask.java

```diff
@@ -76,6 +76,13 @@ public void start(Map<String, String> props, JestClient client) {
 
       if (client != null) {
         this.client = client;
+      } else if (config.getList(ElasticsearchSinkConnectorConfig.CONNECTION_USERNAME)) {
```

Comment:
Username is listed as `Type.STRING` in the config, so this should be `config.getString(...)`.